ethkey/src/secret.rs

```diff
@@ -33,14 +33,21 @@ impl fmt::Debug for Secret {
 }
 
 impl Secret {
-	pub fn from_slice(key: &[u8]) -> Result<Self, Error> {
-		if key.len() != 32 {
-			return Err(Error::InvalidSecret);
-		}
+	fn from_slice_unchecked(key: &[u8]) -> Self {
+		assert_eq!(32, key.len(), "Caller should provide 32-byte length slice");
 
 		let mut h = H256::default();
 		h.copy_from_slice(&key[0..32]);
-		Ok(Secret { inner: h })
+		Secret { inner: h }
+	}
+
+	pub fn from_slice(key: &[u8]) -> Result<Self, Error> {
+		let secret = key::SecretKey::from_slice(&super::SECP256K1, key)?;
+		Ok(secret.into())
+	}
+
+	pub fn from_hash(key: &H256) -> Result<Self, Error> {
+		Self::from_slice(&**key)
```

Comment:
Is this function necessary? From an abstract standpoint getting a secret from a hash doesn't make a whole lot of sense and from a code standpoint the `H256` already has a `Deref` impl.